test/test_smartfile.py

```diff
@@ -0,0 +1,50 @@
+import os
+import unittest
+
+from smartfile import BasicClient
+
+API_KEY = os.environ.get("API_KEY")
+API_PASSWORD = os.environ.get("API_PASSWORD")
+
+if API_KEY is None:
+    raise RuntimeError("API_KEY is required")
+
+if API_PASSWORD is None:
+    raise RuntimeError("API_PASSWORD is required")
+
+
+class CustomOperationsTestCase(unittest.TestCase):
+
+    def setUp(self):
+        self.current_dir = os.path.dirname(os.path.realpath(__file__)) + "/"
+        self.api = BasicClient(API_KEY, API_PASSWORD)
+        self.txtfile = self.current_dir + "myfile.txt"
+        self.uploaddata = None
+
+    def get_data(self):
+        self.uploaddata = self.api.get("/path/info/myfile.txt")
+        return self.uploaddata
+
+    def upload(self):
+        data = open(self.txtfile, "rb")
+        self.api.upload('myfile.txt', data)
```

Comment:
Good job passing the upload method a file object instead of just having the user specify the file name.